src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -186,7 +186,7 @@ internal static SafeFileHandle OpenReadOnly(string fullPath, FileOptions options
         {
             long fileLength;
             Interop.Sys.Permissions filePermissions;
-            return Open(fullPath, mode, access, share, options, preallocationSize, openPermissions, out fileLength, out filePermissions, null);
+            return Open(fullPath, mode, access, share, options, preallocationSize, openPermissions, out fileLength, out filePermissions, createOpenException);
```

Comment:
Are we missing tests then?

Reply:
It causes `File.Copy` with dst being a Directory to have `IOException` message `The file '{0}' already exists.` instead of `The target file '{0}' is a directory, not a file.`.

It regressed in https://github.com/dotnet/runtime/pull/59695 on `main`. It's not in `6.0`.

Reply:
There's a specific code path to get this message. There is no test for it.

Reply:
Ok, so we're exercising the code path and validating we get an exception, we're just not validating the message itself.  That's fine. Thanks.

Reply:
@stephentoub I was wondering whether we should add another assert extension that would allow for verifying exception type and message (but only for `en-*` and invariant cultures). What do you think?

Reply:
We could experiment with it if you think it's valuable.